resident/resident-service/src/main/java/io/mosip/resident/service/impl/OtpManagerServiceImpl.java

```diff
@@ -85,34 +86,28 @@ public boolean sendOtp(MainRequestDTO<OtpRequestDTOV2> requestDTO, String channe
         NotificationRequestDto notificationRequestDto = new NotificationRequestDtoV2();
         notificationRequestDto.setId(identityService.getResidentIndvidualId());
         String refId = this.hash(userId+requestDTO.getRequest().getTransactionID());
-        if (this.otpRepo.checkotpsent(refId, "active", DateUtils.getUTCCurrentDateTime()) > 0) {
+        if (this.otpRepo.checkotpsent(refId, "active", DateUtils.getUTCCurrentDateTime(), DateUtils.getUTCCurrentDateTime()
+                .minusMinutes(this.environment.getProperty("otp.request.flooding.duration", Long.class))) >
+        this.environment.getProperty("otp.request.flooding.max-count", Integer.class)) {
             this.logger.error("sessionId", this.getClass().getSimpleName(), ResidentErrorCode.OTP_ALREADY_SENT.getErrorCode(), "OTP_ALREADY_SENT");
             throw new ResidentServiceCheckedException(ResidentErrorCode.OTP_ALREADY_SENT.getErrorCode(), ResidentErrorCode.OTP_ALREADY_SENT.getErrorMessage());
```

Comment:
Throw innumerous otp error message as done in ida

Reply:
Done.